modules/rest/src/main/java/org/apache/ignite/internal/rest/RestComponent.java

```diff
@@ -99,37 +104,77 @@ public RestComponent(List<RestFactory> restFactories, RestConfiguration restConf
 
-        for (int portCandidate = desiredPort; portCandidate <= desiredPort + portRange; portCandidate++) {
-            try {
-                port = portCandidate;
-                context = buildMicronautContext(portCandidate)
-                        .deduceEnvironment(false)
-                        .environments(BARE_METAL)
-                        .start();
-                LOG.info("REST protocol started successfully");
+        while (httpPortCandidate <= desiredHttpPort + portRange
+                && httpsPortCandidate <= desiredHttpsPort + httpsPortRange) {
+            if (startHttpServer(httpPortCandidate, httpsPortCandidate)) {
                 return;
-            } catch (ApplicationStartupException e) {
-                BindException bindException = findBindException(e);
-                if (bindException != null) {
-                    LOG.debug("Got exception during node start, going to try again [port={}]", portCandidate);
-                    continue;
-                }
-                throw new RuntimeException(e);
+            }
+
+            LOG.debug("Got exception during node start, going to try again [httpPort={}, httpsPort={}]",
+                    httpPortCandidate,
+                    httpsPortCandidate);
+
+            if (sslEnabled && dualProtocol) {
+                httpPortCandidate++;
+                httpsPortCandidate++;
+            } else if (sslEnabled) {
+                httpsPortCandidate++;
+            } else {
+                httpPortCandidate++;
             }
         }
 
-        LOG.debug("Unable to start REST endpoint. All ports are in use [ports=[{}, {}]]", desiredPort, (desiredPort + portRange));
+        LOG.debug("Unable to start REST endpoint."
+                        + " Couldn't find available port for HTTP or HTTPS"
+                        + " [HTTP ports=[{}, {}]],"
+                        + " [HTTP ports=[{}, {}]]",
+                desiredHttpPort, (desiredHttpPort + portRange),
+                desiredHttpsPort, (desiredHttpsPort + httpsPortRange));
 
-        String msg = "Cannot start REST endpoint. " + "All ports in range [" + desiredPort + ", " + (desiredPort + portRange)
-                + "] are in use.";
+        String msg = "Cannot start REST endpoint."
+                + " Couldn't find available port for HTTP or HTTPS"
+                + " [HTTP ports=[" + desiredHttpPort + ", " + desiredHttpPort + portRange + "]],"
+                + " [HTTPS ports=[" + desiredHttpsPort + ", " + desiredHttpsPort + httpsPortRange + "]]";
 
         throw new RuntimeException(msg);
     }
 
+    /** Starts Micronaut application using the provided ports.
+     *
+     * @param httpPortCandidate HTTP port candidate.
+     * @param httpsPortCandidate HTTPS port candidate.
+     * @return {@code True} if server was started successfully, {@code False} if couldn't bind one of the ports.
+     */
+    private boolean startHttpServer(int httpPortCandidate, int httpsPortCandidate) {
```

Comment:
Could you rename it like startServer or startRestServer

Reply:
done